src/fidesops/service/connectors/saas_connector.py

```diff
@@ -0,0 +1,186 @@
+import logging
+from typing import Any, Dict, List, Optional, Tuple
+from requests import Session, Request, PreparedRequest, Response
+
+from fidesops.service.connectors.base_connector import BaseConnector
+from fidesops.graph.traversal import Row, TraversalNode
+from fidesops.models.connectionconfig import ConnectionTestStatus
+from fidesops.models.policy import Policy
+from fidesops.models.privacy_request import PrivacyRequest
+from fidesops.common_exceptions import ConnectionException
+from fidesops.models.saasconfig import SaaSConfig
+from fidesops.models.saasconnectionconfig import SaaSConnectionConfig
+from fidesops.models.saasconfig import ClientConfig
+
+from fidesops.service.connectors.query_config import SaaSQueryConfig
+
+logger = logging.getLogger(__name__)
+
+
+class AuthenticatedClient:
+    """
+    A helper class to build authenticated HTTP requests based on
+    authentication and parameter configurations
+    """
+
+    def __init__(self, secrets: Dict, client_config: ClientConfig):
+        self.s = Session()
+        self.secrets = secrets
+        self.client_config = client_config
+
+    def addAuthentication(
+        self, req: PreparedRequest, strategy_name: str
+    ) -> PreparedRequest:
+        """Uses the incoming strategy to add the appropriate authentication method to the base request"""
+        # TODO: keeping this simple for now since we only have two auth methods
+        configuration = self.client_config.authentication.configuration
+        if strategy_name == "basic_authentication":
```

Comment:
Would it be "overengineering" to use the strategy pattern we're currently using for masking strategies?